File: https:/github.com/rajesh-kayal-dev/Leetcode.git/1-TwoSum/1-TwoSum.js
// Last updated: 6/15/2026, 10:03:50 AM
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map();

    for(let i = 0; i<nums.length;i++){
       let complement =  target - nums[i];
       if(map.has(complement)) {
        return [map.get(complement), i];
       }
       
       map.set(nums[i], i);
    }
    return [];
};

File: https:/github.com/rajesh-kayal-dev/Leetcode.git/1-TwoSum/1-TwoSum.cpp
// Last updated: 6/15/2026, 10:03:52 AM
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int x) {
        int ans[2];
        bool flag=false;
        for(int i =0;i<arr.size();i++){
            for(int j =i+1;j<arr.size();j++){
                if(arr[i]+arr[j]==x){
                   return{i, j};
                    flag=true;
                    break;
                }
            }
        }
        return{};
        
    }
};